src/code/z_player_lib.c

```diff
@@ -742,12 +742,17 @@ void func_8008EE08(Player* this) {
     func_8008EDF0(this);
 }
 
-void func_8008EEAC(PlayState* play, Actor* actor) {
+/**
+ * Sets the "auto lock-on actor" to lock onto an actor without Player's input.
+ * This function will first release any existing lock-on or (try to) release parallel.
+ * The new lock-on is considered "friendly" even if the actor is actually hostile.
```

Comment:
is it intended that it uses "friendly" lock-on? what effects does this have?

Reply:
Well for one player wont have his "hostile" battle response. but this doesnt matter in both contexts this is used anyway cause he is either frozen or in a chokehold.

I think more importantly it will use the "focus" or "keepon" camera mode instead of "battle". Which just leads to a slightly different looking camera.

Reply:
will the camera switch back to being "hostile" after the chokehold is released? or is it permanently friendly?

Reply:
So it seems like with switch targeting you cannot keep the lockon held after an auto lockon ends. It will release automatically. This is likely because `PLAYER_STATE2_LOCK_ON_WITH_SWITCH` never gets set.

You can keep the lock-on held with hold targeting however. And it indeed switches from "friendly" to "hostile" when the auto lockon is over.

See clip below with the camera debugger enabled. "KEEPON" is friendly actor focus and "BATTLE" is hostile.

https://github.com/user-attachments/assets/fd8100ab-29f1-4efd-88db-3120a57b9e73

Reply:
oh interesting, thanks for looking. Maybe copy-paste that into the comment?

Reply:
Sure.

I think general speaking what is currently called the `CAM_MODE_Z_TARGET_FRIENDLY` mode is really just a general actor focus mode (`CAM_MODE_ACTOR_FOCUS`). Nothing it particularly "friendly" about it.

However `PLAYER_STATE1_FRIENDLY_ACTOR_FOCUS` being set would make Link unable to have his battle response, if he were actionable during that time.

Not sure how in depth to go here with the comment but I will try to make it reasonable